echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/TriggerMonitor.java

```diff
@@ -67,12 +69,16 @@ private void validateEvent(Event event) {
   }
 
   private void triggerMatchingPipelines(T event) {
-    List<Pipeline> allPipelines = pipelineCache.getPipelinesSync();
-    List<Pipeline> matchingPipelines = eventHandler.getMatchingPipelines(event, allPipelines);
-    matchingPipelines.forEach(p -> {
-      recordMatchingPipeline(p);
-      pipelineInitiator.startPipeline(p);
-    });
+    try {
+      List<Pipeline> allPipelines = pipelineCache.getPipelinesSync();
+      List<Pipeline> matchingPipelines = eventHandler.getMatchingPipelines(event, allPipelines);
+      matchingPipelines.forEach(p -> {
+        recordMatchingPipeline(p);
+        pipelineInitiator.startPipeline(p);
+      });
+    } catch (TimeoutException e) {
```

Comment:
I'm wondering if logging and continuing is the right fallback here?  Ideally we'd probably want to leave the event unprocessed rather than ignore it, but as echo doesn't really have a queuing mechanism, I suppose that failing is probably the only option.

Reply:
Yes, ideally we wouldn't run into this in the first place. I'd like to find a way to key some kind of health indicator based on the pipeline cache status. E.g. cache not initialized, echo is not "healthy" since it's not ready to do the work.

I'm afraid that queueing would compound ongoing issues and make the system more fragile overall